test/infrastructure/docker/api/v1beta1/dockercluster_types.go

```diff
@@ -53,6 +54,13 @@ type DockerClusterSpec struct {
 type DockerLoadBalancer struct {
 	// ImageMeta allows customizing the image used for the cluster load balancer.
 	ImageMeta `json:",inline"`
+
+	// AdditionalProxyConfigRef allows to add additional configuration to the HAProxy config file.
+	// This field is a reference to a config map that contains the extra configuration. The key of the config map
+	// should be equal to 'value'.
+	// The content of the config map will be appended to the HAProxy config file. Please use it with caution,
+	// as there are no checks to ensure the validity of the configuration.
+	AdditionalProxyConfigRef *corev1.LocalObjectReference `json:"additionalProxyConfigRef,omitempty"`
```

Comment:
What about
```suggestion
	UnsafeHAProxyConfigTemplateRef *corev1.LocalObjectReference `json:"additionalProxyConfigRef,omitempty"`
```
So we are explicity about "use at your own risk"; also, dropped additional according to latest discussion in the issue (it should be replace)